lib/KBox/src/KMessage.h

```diff
@@ -191,6 +196,94 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage {
+  private:
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint;
```

Comment:
Finally, I think your code would greatly benefit if you replaced your 4 mode variables with one enum that lists all the possible states of the autopilot. There are so many variables now that it is hard to verify that we are not missing a case in the `if` statements that do logic on them.